examples/c/mnist_slp_symbol.c

```diff
@@ -0,0 +1,49 @@
+#include <assert.h>
+
+#include <crystalnet.h>
+
+// y = softmax(flatten(x) * w + b)
+s_model_t *slp(shape_t *image_shape, uint8_t arity)
+{
+    shape_t *lable_shape = make_shape(1, arity);
```

Comment:
Is it really necessary to ask users to new/free shape here? 

In case of creating a complex neural network, this would largely complicate the code.

Reply:
one way of auto `free_shape` is let `make_shape` take a `shape_ctx_t *` parameter, which manages the ownerships of shape_t:
```
shape_t = make_shape(ctx, 2, w, b);   // it calls ctx->make_shape(...);
...
```

in `shape_ctx_t`:

```
struct shape_ctx_t {
    template <typename ... T>
    shape_t make_shape(T ... dims) {
       return gc(new shape_t(dims...);
    }
};
```

Reply:
implemented as https://github.com/lgarithm/crystalnet/commit/0800b9ddb86c350c1777f69370ed1cad32779d9b